template/README.md

```diff
@@ -404,6 +405,44 @@ Note that GitHub Pages doesn't support routers that use the HTML5 `pushState` hi
 
 Use the [Heroku Buildpack for create-react-app](https://github.com/mars/create-react-app-buildpack).
 
+### Adding Custom Environment Variables
+
+> Note: this feature is available with `v0.3.0` and higher.
+
+Your project can consume variables declared in your environment as if they were declared locally in your JS files. By
+default you will have `NODE_ENV` defined for you, and any other environment variables starting with
+`REACT_APP_`. These environment variables will be defined for you on `process.env`. For example, having
+`REACT_APP_SECRET_CODE=abcdef` will be exposed in your JS as `process.env.REACT_APP_SECRET_CODE`, along with
+`process.env.NODE_ENV`.
+
+These environment variables can be useful for displaying information conditionally based on where the project is
+deployed or consuming sensitive data that lives outside of version control.
+
+```bash
+$ REACT_APP_SECRET_CODE=abcdef
+$ npm start
+```
+
+```jsx
+<h1>Hello, Admin!<h1>
+
+<small>You are current running this application in <strong>{process.env.NODE_ENV}</strong> mode.</small>
```

Comment:
It is not immediately clear what the user will see. Let’s add a second code block that shows the result.
Ideally it should be like this:

1. Show how to access variables with this code example
2. Then show how to specify env variable outside (separate instructions for OS X, Windows)
3. Then show what the result would look like given those values

This gives the reader the sense of the complete flow.